src/Compilers/CSharp/Test/Semantic/Semantics/WarningVersionTests.cs

```diff
@@ -139,13 +139,69 @@ public Program(int dummy)
     {
     }
 }";
-            CreateCompilation(source2, references: new MetadataReference[] { moduleReference }, options: TestOptions.ReleaseDll.WithWarningLevel(CodeAnalysis.Diagnostic.DefaultWarningLevel)).VerifyDiagnostics(
-                );
-            CreateCompilation(source2, references: new MetadataReference[] { moduleReference }, options: TestOptions.ReleaseDll.WithWarningLevel(5)).VerifyDiagnostics(
-                // (4,12): warning CS8822: Auto-implemented property 'Program.Property' must be fully assigned before control is returned to the caller.
+            var csharp10IL = @"
+{
+  // Code size        2 (0x2)
+  .maxstack  0
+  IL_0000:  nop
+  IL_0001:  ret
+}
+";
+            var csharp11IL = @"
+{
+  // Code size       14 (0xe)
+  .maxstack  1
+  IL_0000:  ldarg.0
+  IL_0001:  ldflda     ""Struct Program.<Property>k__BackingField""
+  IL_0006:  initobj    ""Struct""
+  IL_000c:  nop
+  IL_000d:  ret
+}
+";
+            // C# 10
+            // note: no LangVersion diagnostics are given here, even though we will change codegen in the later language version
+            var verifier = CompileAndVerify(
+                source2,
+                references: moduleReference,
+                options: TestOptions.DebugDll.WithWarningLevel(CodeAnalysis.Diagnostic.DefaultWarningLevel),
+                parseOptions: TestOptions.Regular10,
+                verify: Verification.Skipped);
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("Program..ctor", csharp10IL);
+
+            verifier = CompileAndVerify(
+                source2,
+                references: moduleReference,
+                options: TestOptions.DebugDll.WithWarningLevel(5),
+                parseOptions: TestOptions.Regular10,
+                verify: Verification.Skipped);
+            verifier.VerifyDiagnostics(
+                // (4,12): warning CS8880: Auto-implemented property 'Program.Property' must be fully assigned before control is returned to the caller.
                 //     public Program(int dummy)
-                Diagnostic(ErrorCode.WRN_UnassignedThisAutoProperty, "Program").WithArguments("Program.Property").WithLocation(4, 12)
-                );
+                Diagnostic(ErrorCode.WRN_UnassignedThisAutoProperty, "Program").WithArguments("Program.Property").WithLocation(4, 12));
+            verifier.VerifyIL("Program..ctor", csharp10IL);
+
+            // C# 11+
+            verifier = CompileAndVerify(
+                source2,
+                references: moduleReference,
+                options: TestOptions.DebugDll.WithWarningLevel(5),
+                verify: Verification.Skipped);
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("Program..ctor", csharp11IL);
+
+            verifier = CompileAndVerify(
```

Comment:
Tests where this PR made Regular10 explicit should also be testing RegularNext. Currently, they are testing with "preview", which will keep rolling forward to C# 12, etc.